improver_tests/threshold/test_BasicThreshold.py

```diff
@@ -599,6 +599,15 @@ def test_threshold_point_nan(self):
         with self.assertRaisesRegex(ValueError, msg):
             plugin(self.cube)
 
+    def test_each_threshold_func(self):
+        """Test user supplied func is applied on each threshold cube."""
+        # Need to copy the cube as we're adjusting the data.
+        plugin = Threshold(
+            2.0, each_threshold_func=lambda cube: cube.rename("new_name") or cube
```

Comment:
What is the `or cube` statement doing here?

Reply:
a little trick to return `cube` from `lamda` after some in place operation - `a or b` is the same as `a if a else b` and, since `cube.rename("new_name")` resolves to `None`, the whole expression resolves to `cube`, but only after the rename happened.

Reply:
Ah OK.  Clever.  Figures.